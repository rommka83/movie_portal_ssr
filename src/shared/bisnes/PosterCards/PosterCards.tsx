import React, { HTMLAttributes, useMemo } from 'react';
import styles from './postercards.module.css';
import classNames from 'classnames';
import { SvgIcon } from 'shared/ui/SvgIcon';
import Image from 'next/image';

interface IProps {
  src?: string;
  name: string;
  top?: boolean;
  num?: number;
}
type props = HTMLAttributes<HTMLDivElement> & IProps;

export const PosterCards = ({ src, name, className, top, num }: props) => {
  const number = useMemo(() => {
    const _num = num ? num + 1 : 1;
    return 'number' + _num;
  }, [num]);

  return top ? (
    <div className={classNames(styles.wrapper, className)}>
      {src && <Image width={200} height={300} src={src} alt={name} className={styles.pic} />}
      {number != 'number10' ? (
        <div className={styles.num}>
          <SvgIcon type={number} />
        </div>
      ) : (
        <div className={styles.num}>
          <SvgIcon type={'number1'} />
          <SvgIcon type={'number0'} />
        </div>
      )}
    </div>
  ) : (
    <div className={classNames(styles.wrapper, className)}>
      {src && <Image src={src} width={200} height={300} alt={name} className={styles.pic} />}
    </div>
  );
};
