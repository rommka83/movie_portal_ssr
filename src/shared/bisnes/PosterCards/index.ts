export { PosterCards as default } from './PosterCards';
