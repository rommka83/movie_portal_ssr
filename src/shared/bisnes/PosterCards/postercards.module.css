.wrapper {
  position: relative;
  width: 100%;
  height: 100%;
  border-radius: 8px;
  aspect-ratio: 64 / 91;
  overflow: hidden;
  transform: scale(0.95);
  transition: transform 0.3s ease-in-out;
  background-color: var(--gray-798);
}
.pic {
  width: 100%;
  height: 100%;
  object-fit: cover;
}
.wrapper:hover {
  transform: scale(1);
}
.num {
  position: absolute;
  left: 0;
  bottom: -5%;
  display: flex;
  justify-content: center;
  width: 100%;
  height: 40%;
  background: linear-gradient(0deg, rgb(21, 22, 22), rgba(241, 247, 247, 0));
}
.num > svg {
  width: auto;
  height: 100%;
}
