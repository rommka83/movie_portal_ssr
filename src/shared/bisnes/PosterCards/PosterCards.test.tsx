import renderer from 'react-test-renderer';
import { PosterCards } from './PosterCards';

describe('проверка PosterCards', () => {
  test('отрисовка PosterCards ', () => {
    const component = renderer.create(<PosterCards src={''} name={''} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
