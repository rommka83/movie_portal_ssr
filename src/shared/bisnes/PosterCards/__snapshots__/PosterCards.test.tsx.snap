// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка PosterCards отрисовка PosterCards  1`] = `
<div
  className="wrapper"
/>
`;
